mlir/python/mlir/dialects/transform/__init__.py

```diff
@@ -211,3 +211,7 @@ def __init__(
         if operands is None:
             operands = []
         super().__init__(_get_op_results_or_values(operands), loc=loc, ip=ip)
+
+
+def any_op_t():
```

Comment:
Not familiar with this code and what AnyOpType is, but is that actually a descendent of a Python `type`? If not, it's going to make the various tools unhappy to use it in signatures. (answering my own question: appears not)

I think you need to do an additional indirection here to make things happy. Easiest thing is to use something like `typine.newtype`. iirc, there is also another way to create an annotated type that the type checkers will look through. You can also, of course go full on dependent typing with metaclasses...

Reply:
This is just this is just `transform.AnyOpType` but exposed [through their C extension module](https://github.com/llvm/llvm-project/blob/main/mlir/lib/Bindings/Python/DialectTransform.cpp#L29). None such getters are effectively type hinted except insofar as the generated stub goes (and that's not being done for the various (even upstream) dialect C extensions).

Reply:
Yeah -- what I'm saying is that if you put this like this in user code without playing nice with type checkers, they are going to yell at your users.

Reply:
just double checking I did this right:

```python
AnyOpTypeT = NewType("AnyOpType", AnyOpType)

def any_op_t() -> AnyOpTypeT:
    return AnyOpTypeT(AnyOpType.get())
```